### PR TITLE
[exporter/clickhouse] Implement consume log logic

### PR DESCRIPTION
**Pull request by hanjm**
**Description:** <Describe what has changed.>

This is the second PR to implement clickhouseexporter.
First PR: #6907 

Plan:
- Add e2e test.
- Add more index.
- Support Distributed table.
- Support consume traces.
- Add benchmark.

**Link to tracking Issue:** <Issue number if applicable>
#2320 
**Testing:** <Describe what testing was performed and which tests were added.>
1. Unit test.
2. Test with docker-compose (in `example` dir).

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: exporter/clickhouseexporter/exporter.go
Comment (dmitryax) [inline exporter/clickhouseexporter/exporter.go:205]: Does it mean that Rollback will be executed regardless of the error?

**Comment by dmitryax** [comment]
Please rebase to resolve conflicts

File: exporter/clickhouseexporter/exporter.go
Comment (hanjm) [reply exporter/clickhouseexporter/exporter.go:205]: No, it will return  ErrTxDone if tx is commit https://cs.opensource.google/go/go/+/master:src/database/sql/sql.go;l=2304?q=rollback&ss=go%2Fgo.
It just a common usage, like this example: https://pkg.go.dev/database/sql#example-Tx.Prepare

File: exporter/clickhouseexporter/exporter.go
Comment (dmitryax) [inline exporter/clickhouseexporter/exporter.go:138]: I would recommend to call the column `ResourceAttributes`, but I leave it up to you

File: exporter/clickhouseexporter/README.md
Comment (dmitryax) [inline exporter/clickhouseexporter/README.md:28]: Why `logs_local`? If we stick to Otel data model, maybe better to call it `otel_logs`? Also wouldn't user be interested to use a custom table name?

Comment (hanjm) [reply exporter/clickhouseexporter/README.md:28]: Make sense.

File: exporter/clickhouseexporter/exporter.go
Comment (hanjm) [reply exporter/clickhouseexporter/exporter.go:138]: This field name is align to the specification. I want to keep it. thanks. https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md#log-and-event-record-definition

File: exporter/clickhouseexporter/README.md
Comment (hanjm) [reply exporter/clickhouseexporter/README.md:28]: OK. i will update it today. :)

File: exporter/clickhouseexporter/exporter.go
Comment (dmitryax) [reply exporter/clickhouseexporter/exporter.go:138]: Hm. The definition of the Resource is not correct in the doc. Resource its not the same as Log.Attributes. Resource has Attributes as a field: Resource.Attributes. Looks at the proto definition https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/resource/v1/resource.proto#L31

Comment (dmitryax) [reply exporter/clickhouseexporter/exporter.go:138]: It think it would be more clear to call them LogAttributes (or LogAttrs) and ResourceAttributes (or ResourceAttrs).

File: exporter/clickhouseexporter/README.md
Comment (hanjm) [reply exporter/clickhouseexporter/README.md:28]: Update done.

File: exporter/clickhouseexporter/exporter.go
Comment (hanjm) [reply exporter/clickhouseexporter/exporter.go:142]: I See. I can change
```
    Attributes Nested
        (
        Key LowCardinality(String),
        Value String
        ) CODEC(ZSTD(1)),
    Resource Nested
        (
        Key LowCardinality(String),
        Value String
        ) CODEC(ZSTD(1)),
```
to
```
    LogAttributes Nested
        (
        Key LowCardinality(String),
        Value String
        ) CODEC(ZSTD(1)),
    ResourceAttributes Nested
        (
        Key LowCardinality(String),
        Value String
        ) CODEC(ZSTD(1)),
```
, right?

Comment (dmitryax) [reply exporter/clickhouseexporter/exporter.go:142]: Yes, looks correct

Comment (hanjm) [reply exporter/clickhouseexporter/exporter.go]: Done.

**Comment by hanjm** [comment]
Rebase done again. Thank you.

**Review by dmitryax — APPROVED** [review]
LGTM

**Comment by hanjm** [comment]
@dmitryax  Some test flaky 😢 , what could I do? 

Another excited message is the latest grafana-clickhouse-datasource support query clickhouse logs in explore UI.  https://github.com/grafana/clickhouse-datasource/issues/23

**Comment by hanjm** [comment]
hi @dmitryax @mx-psi , rebase done and all ci checkers passed.

**Comment by dmitryax** [comment]
@hanjm thanks. One of the @open-telemetry/collector-contrib-maintainer can merge it

**Review by codeboten — APPROVED** [review]

**Pull request merged**
